src/main/java/htsjdk/samtools/cram/structure/block/Block.java

```diff
@@ -66,15 +71,18 @@
      *
      * @param compressionMethod the block compression method.  Can be RAW, if uncompressed
      * @param contentType whether this is a header or data block, and which kind
+     * @param contentId the External Block Content ID, or NO_CONTENT_ID for non-External block
      * @param compressedContent the compressed form of the data to be stored in this block
      * @param uncompressedLength the length of the content stored in this block when uncompressed
      */
     protected Block(final BlockCompressionMethod compressionMethod,
                     final BlockContentType contentType,
+                    final int contentId,
                     final byte[] compressedContent,
                     final int uncompressedLength) {
         this.compressionMethod = compressionMethod;
         this.contentType = contentType;
+        this.contentId = contentId;
```

Comment:
We should add a unit test in `BlockTest` with a data provider for all of the block types, and a negative test that verifies that this method won't create a block with a content ID unless its an external block.